nf_core/modules/lint/main_nf.py

```diff
@@ -268,6 +269,12 @@ def check_process_section(self, lines, fix_version, progress_bar):
             else:
                 self.failed.append(("singularity_tag", "Unable to parse singularity tag", self.main_nf))
                 singularity_tag = None
+            url_regex = (
```

Comment:
Why not use `urllib.parse.urlparse`? I think it might help to reduce the regex gibberish at least partially.

Reply:
thanks for the suggestion :)

Reply:
So much nicer! 🤩